app/models/nearby_facility.rb

```diff
@@ -82,6 +87,26 @@ def empty_resource_set?(response_body)
       response_body['resourceSets'].size.zero? || response_body['resourceSets'][0]['estimatedTotal'].zero?
     end
 
+    def waypoint(params)
+      type = location_type(params)
+      values = REQUIRED_PARAMS[type].map { |k| params[k].to_s }
+      case type
+      when :address
+        return values.join(' ')
+      when :lat_lng
+        return values.join(',')
+      else
+        return ''
+      end
+    end
+
+    def location_type(params)
+      REQUIRED_PARAMS.each do |key, value|
+        return key if (value - params.keys.map(&:to_sym)).empty?
+      end
+      nil
+    end
```

Comment:
`find` might be a bit more idiomatic
```
    def location_type(params)
      REQUIRED_PARAMS.find do |key, value|
        break key if (value - params.keys.map(&:to_sym)).empty?
      end
    end
```

Reply:
Avoiding opaque variable names and relying on the principle of one-concept-per-line, I'd write it like this:

```
    def location_type(params)
      obs_fields = params.keys.map(&:to_sym)                           # only do this once
      REQUIRED_PARAMS.find do |loc_type, req_field_names|              # more meaningful variable names
        any_missing_fields = (req_field_names - obs_fields).empty?     # give this list a meaningful name
        any_missing_fields and loc_type                                # rely on the find method false-y semantics
      end
    end
```

(comments just for this PR)

Reply:
@dvogel will your use of `find` return what he wants? you can break from the block and return anything you want, but, if you let the block naturally return (if that makes sense), it will return an array of [key, value] (in this scenario (find on a hash)) 
```
{a: 1, b: 2}.find { |k,v| v % 2 == 0} # [:b, 2]


{a: 1, b: 2}.find { |k,v| break "#{v} is even!" if v % 2 == 0} # "2 is even!"
```
maybe i'm misunderstanding

Reply:
@edmangimelli you're right, I overlooked that bit. Using the first 3 comments as my rationale,  I would still write the last line of the block as:

```
any_missing_fields and break loc_type
```

Or, as applied to your example:

```
{a: 1, b: 2}.find { |k,v| (v % 2 == 0) and break "#{v} is even!" }
```